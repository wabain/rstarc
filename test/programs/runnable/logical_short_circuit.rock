(Test value assignment via logical short-circuit)

put "some value" and false into AndAssignment
say "Assignment via and (expected: some value)"
say AndAssignment

put false or "some value" into OrAssignment
say "Assignment via or (expected: some value)"
say OrAssignment

put "yes" nor "no" into NorAssignment
say "Assignment via nor (expected: false)"
say NorAssignment

put 0 nor "no" into NorAssignment
say "Assignment via nor (expected: true)"
say NorAssignment
