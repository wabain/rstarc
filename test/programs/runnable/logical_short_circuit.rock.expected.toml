[tests]
enabled = ["run"]

[run]
returncode = "0"
stdout = """
Assignment via and (expected: some value)
some value
Assignment via or (expected: some value)
some value
Assignment via nor (expected: false)
false
Assignment via nor (expected: true)
true
"""
